packages/e2e-tests/specs/editor/various/post-editor-template-mode.test.js

```diff
@@ -197,3 +197,167 @@ describe( 'Post Editor Template mode', () => {
 		expect( content ).toMatchSnapshot();
 	} );
 } );
+
+describe( 'Delete Post Template Confirmation Dialog', () => {
+	beforeAll( async () => {
+		await activatePlugin( 'gutenberg-test-block-templates' );
+		await trashAllPosts( 'wp_template' );
+		await trashAllPosts( 'wp_template_part' );
+		await activateTheme( 'twentytwentyone' );
+		await createNewPost();
+		// Create a random post.
+		await page.type( '.editor-post-title__input', 'Just an FSE Post' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'Hello World' );
+
+		// Save the post
+		// Saving shouldn't be necessary but unfortunately,
+		// there's a template resolution bug forcing us to do so.
+		await saveDraft();
+		await page.reload();
+		// Unselect the blocks.
+		await page.evaluate( () => {
+			wp.data.dispatch( 'core/block-editor' ).clearSelectedBlock();
+		} );
+	} );
+
+	afterAll( async () => {
+		await activateTheme( 'twentytwentyone' );
+		await deactivatePlugin( 'gutenberg-test-block-templates' );
+	} );
+
+	[ 'large', 'small' ].forEach( ( viewport ) => {
+		it( `should retain template if deletion is canceled when the viewport is ${ viewport }`, async () => {
+			await setBrowserViewport( viewport );
+
+			const isWelcomeGuideActive = await page.evaluate( () =>
+				wp.data
+					.select( 'core/edit-post' )
+					.isFeatureActive( 'welcomeGuide' )
+			);
+			if ( isWelcomeGuideActive === true ) {
+				await page.evaluate( () =>
+					wp.data
+						.dispatch( 'core/edit-post' )
+						.toggleFeature( 'welcomeGuide' )
+				);
+				await page.reload();
+				await page.waitForSelector( '.edit-post-layout' );
+			}
+			const templateTitle = `${ viewport } Viewport Deletion Test`;
+
+			await createNewTemplate( templateTitle );
+			// Edit the template
+			if ( viewport === 'small' ) {
+				const closeDocumentSettingsButton = await page.waitForXPath(
+					'//div[contains(@class,"interface-complementary-area-header__small")]/button[@aria-label="Close settings"]'
+				);
+				await closeDocumentSettingsButton.click();
+			}
+			await insertBlock( 'Paragraph' );
+			await page.keyboard.type(
```

Comment:
The previous test in the suite leaves the site on Twenty Twenty-one, which works well for the tests I've added. I did notice, however that switching to another theme (tested Twenty Twenty-two and Emptytheme) causes the new tests to fail (see line 206 for the e2e-util that handles the theme switch)

On either of those two themes, and potentially others, this call to `page.keyboard.type` isn't able to complete. It stops midway through the string for reasons I can't see and the test ends up failing.